modin/experimental/core/execution/ray/implementations/pyarrow_on_ray/partitioning/axis_partition.py

```diff
@@ -36,7 +36,7 @@ class PyarrowOnRayDataframeAxisPartition(BaseDataframeAxisPartition):
 
     def __init__(self, list_of_blocks):
         # Unwrap from PandasDataframePartition object for ease of use
-        self.list_of_blocks = [obj._data for obj in list_of_blocks]
+        self.list_of_blocks = [obj.list_of_blocks[0] for obj in list_of_blocks]
```

Comment:
```suggestion
        self.list_of_blocks = [block for partition in list_of_blocks for block in partition.list_of_blocks]
```

I think you could initialize this class with a list including virtual partitions?

Reply:
I'm not sure that it will be correct. In this case len(`self.list_of_blocks`) might be different. Are we breaking the correct partitioning of `PyarrowOnRayDataframeAxisPartition` here or not?

Reply:
I guess we should assume these are always "full axis" partitions that are made of other `PyarrowOnRayDataframeAxisPartition`. Let's add another assertion here that all `list_of_blocks` have length 1.